detox/src/Detox.js

```diff
@@ -109,14 +78,19 @@ class Detox {
 
     if (this._server) {
       await this._server.close();
+      this._server = null;
     }
 
     if (argparse.getArgValue('cleanup') && this.device) {
       await this.device.shutdown();
     }
+
+    this.device = null;
   }
 
   async beforeEach(testSummary) {
+    await this._assertNoPendingInit();
```

Comment:
Thank you, `jest-jasmine` :nauseated_face: :rage:  .
If test runner proceeds to run individual tests with their hooks after a timeout, `beforeEach` now will be mercilessly throwing errors.

Reply:
So this is here in order to handle timeouts in the global init, in particular?